### ISSUE TITLE
[Bug Report] v-skeleton-loader breaks v-data-table with server side data load

### ISSUE THREAD

**Issue by poganytamastsys** [vuetifyjs/vuetify#9357]
### Environment
**Vuetify Version:** 2.1.3
**Vue Version:** 2.6.10
**Browsers:** Chrome 77.0.3865.90
**OS:** Windows 10

### Steps to reproduce
Construct a data table with server side data loading. Put it inside a v-skeleton-loader (or use the v-if, doesn't matter).

### Expected Behavior
The skeleton loader should disappear after server data loaded.

### Actual Behavior
Loading stays true, server data never gets loaded.

### Reproduction Link
<a href="https://codepen.io/poganytamastsys/pen/OJJNZpb" target="_blank">https://codepen.io/poganytamastsys/pen/OJJNZpb</a>

### Other comments
After removing the skeleton loader, it starts to work immediately.

**Comment by DRoet** [comment]
Your codepen has some errors:

1) `sortBy` and `sortDesc` are both undefined (check the console to see the error)
2) The watcher is looping indefinitely, this is why `this.loading` always stays `true`

**Comment by zayats103** [comment]
@DRoet I have the same problem. Here is a link - https://codepen.io/sanyazayats/pen/BaareeR (click `Run` the pen after it is loaded and check console). All the code is copy-pasted from Data-table example https://vuetifyjs.com/en/components/data-tables and just skeleton loader was added.
Why watcher is looping indefinitely? If you remove skeleton there will be no indefinite loop of watcher. Looks like skeleton is changing the `options` and that's why indefinite loop appears.

**Comment by poganytamastsys** [comment]
> . Looks like skeleton is changing the `options` and that's why indefinite loop appears.

And you are absolutely correct. It has something to do with data-table not being rendered while the skeleton is loading. I ended up in my case with using the loading animation of data-table, and forgot the skeleton. Could not figure out the watcher, because the properties are the same for old and new value. Maybe javascript can still detect it is a completely new object, even the property values are the same.

**Comment by poganytamastsys** [comment]
One workaround is to include the options object inside the loading property of the skeleton.
Something like this: [https://codepen.io/poganytamastsys/pen/ZEEoyzM?editors=1111](https://codepen.io/poganytamastsys/pen/ZEEoyzM?editors=1111)

Not elegant, and maybe there is a better solution, but it works.

**Comment by zayats103** [comment]
@poganytamastsys Yes, with such "hack" it does work, but in console `data is loaded` is printed twice. In case of real API call for fetching data it will cause unnecessary additional calls

**Comment by nekosaur** [comment]
This happens because `loading` is set to `true` every time in the `getDataFromApi` method. That means the skeleton loader will remove the table every time, and then re-render it, cause the options object to get updated, causing the handler to call `getDataFromApi`, rinse & repeat. What you want to do is something like this https://codepen.io/nekosaur/pen/MWWGxRd

Where you have a separate variable just for the initial load

**Comment by zayats103** [comment]
@nekosaur Thank you for explanation and clarification. But in your example there is no point of skeleton since information is loaded once again after skeleton dissapear (since options are changed). It is better to neither fetch data in `mounted` method nor use skeleton's slots, but use `v-show` for the grid instead. In this case grid will be hidden, but will load all the data and initialize itself while skeleton is shown (using suggested `firstLoad` trick). Furthermore (since we don't load data in `mounted` method) there will be only one call of `getDataFromApi` method since it will be triggered only from watcher. See my updated pen https://codepen.io/sanyazayats/pen/BaareeR
